### ISSUE TITLE
Reset is not localized to the current namespace

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#252]
Test program (perl 522 with debug enabled)
````
> cat test.pl
package bar;
sub search { shift =~ m?bar? ? 1 : 0 }
sub reset_zlopp { reset }

package foo;
sub ZZIP { shift =~ m?ZZIP? ? 1 : 0 }

package main;

foo::ZZIP("ZZIP");
bar::reset_zlopp();
print !foo::ZZIP("ZZIP") ? "ok\n" : "not ok\n";
````

When resetting the smartmatch for bar, foo is also reset

````
> perl test.pl
ok
> perlcc -r test.pl
/usr/local/cpanel/3rdparty/perl/522-debug/bin/perlcc: Unexpected compiler output
Use of uninitialized value $pmop_ptr in hash element at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3257.
 Use of uninitialized value $pmop_ptr in sprintf at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3258.
 pmop 0x0 not found in our B::C Regexp hash at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3258.
 Use of uninitialized value $pmop_ptr in hash element at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3257.
 Use of uninitialized value $pmop_ptr in sprintf at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3258.
 pmop 0x0 not found in our B::C Regexp hash at /usr/local/cpanel/3rdparty/perl/522-debug/lib/perl5/cpanel_lib/i386-linux-debug-64int/B/C.pm line 3258.
pccNxtPu.c: In function ‘perl_init_aaaa’:
pccNxtPu.c:402: warning: assignment from incompatible pointer type
pccNxtPu.c:473: warning: assignment from incompatible pointer type
not ok
````

**Comment by toddr** [comment]
./t/testc.sh 188

**Comment by atoomic** [comment]
./testc.sh 2741is very close to this test, needs to be adjusted

**Comment by toddr** [comment]
perlcc -O3 -r -e 'package bar; sub search { shift =~ m?bar? ? 1 : 0 } sub reset_zlopp { reset } package foo; sub ZZIP { shift =~ m?ZZIP? ? 1 : 0 } package main; foo::ZZIP("ZZIP"); bar::reset_zlopp(); !foo::ZZIP("ZZIP") and print "ok\n"'

**Comment by toddr** [comment]
./t/testc.sh 252

**Comment by rurban** [comment]
The problem is rather that `PmopSTASHPV((PMOP*)PL_op)` in `C.xs` is now empty, and thus the MATCH once op is not stored in `%B::C::Regexp`, causing the very first MATCH to be used instead of the correct one.

**Comment by rurban** [comment]
Fixed with 8fece2f
With 5.20 our own runloop does not find all PMOPs anymore
